app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -2,6 +2,7 @@ module OpenidConnect
   class AuthorizationController < ApplicationController
     include FullyAuthenticatable
     include VerifyProfileConcern
+    include VerifySPAttributesConcern
```

Comment:
It looks like the OIDC authorization controllers and SAML IDP controllers could still use some cleanup. I think it would be a bit more readable if we didn’t mix the verify/profile redirects with the attribute verification redirects. It would be nice to see the verification redirects as separate guard clauses, maybe with a new `handle_successful_handoff` method or similar at the end if rubocop or reek gives you trouble.

```ruby
def index
  return confirm_two_factor_authenticated(request_id) unless user_fully_authenticated?
  @authorize_form.link_identity_to_service_provider(current_user, session.id)
  return redirect_to_account_or_verify_profile_url if profile_or_identity_needs_verification?
  return redirect_to(sign_up_completed_url) unless needs_sp_attributes_verification
  handle_successful_handoff
```

You should be able to do something similar with the SAML IDP controller, albeit with some more deconstruction. Happy to pair with you on this if you want.